cli/ingress-controller/flags.go

```diff
@@ -311,6 +319,9 @@ func parseFlags() (cliConfig, error) {
 	config.LogLevel = viper.GetString("log-level")
 	config.LogFormat = viper.GetString("log-format")
 
+	// Diagnostics
+	config.DumpConfig = viper.GetString("dump-config")
```

Comment:
`cfg.DumpConfig` _behaves like_ an enumeration type, and I think it'd make sense to implement it as such:

```go
type ConfigDumpMode int

const (
	ConfigDumpModeOff ConfigDumpMode = iota
	ConfigDumpModeEnabled ConfigDumpMode = iota
	ConfigDumpModeSensitive ConfigDumpMode = iota
)
```

with a function like `func parseConfigDumpMode(in string) (ConfigDumpMode, error)` being run here, to parse the `string` flag into a `DumpConfigMode`, and failing (exiting) if the value doesn't parse.